### PR TITLE
Done: Refactor shedtools

### PR DESCRIPTION
**Pull request by rhpvorderman**
~~After two attempts refactoring the existing file, I am going to refactor shed tools from scratch. I will use the issue #91 as a guideline.~~

~~Please do not add any functionality to shed tools before the refactoring complete. I will be pushing regularly so you can check on my progress. I will also be available on gitter.~~

Done.
The following stuff has changed:
+ shed_tools arguments and methods now in a separate file (for readability reasons)
+ InstallToolManager rewritten to allow for a much cleaner interface. No storing of variables in self that are used by only one method. InstallToolManager can now be instantiated with only a GalaxyInstance. Methods also have a sane number of arguments. So, when people where referencing shed_tools.py in other scripts, these scripts are now broken. On the upside, referencing shed_tools in other scripts is now much easier
+ Global log was removed
+ Complexity dropped by 3 units. (Whatever that is supposed to mean)
+ Useless stuff was chucked out. Some defaults were passed through functions 5 times.
+ get-tool-list can now get all the revisions of each tool, not just the revisions that are in the tool panel. Admin rights are needed for this. Shed-tools uses this functionality to get the installed tools.
+ Whether tools are already installed is now determined by getting a list of the installed tools and checking that. The previous method did this via checking every tool with an API call to galaxy. This was of course slow. Now it should be faster (even though the checking method is gruesomely inefficient, but it should be faster than doing ssl handshakes all the time).
+ `shed-tools update` now also accepts tool lists. It will only update tools that are already installed on the galaxy and issue a warning for other tools.
+ The `--skip_install_tool_dependencies` flag is no longer functional as this is the default. Same goes for the `--install_resolver_dependencies` flag. These flags are still around for backwards compatibility, but do not do anything. `--install_tool_dependencies`, `--skip_install_resolver_dependencies` and `--skip_install_repository_dependencies` where added.
+ Probably some stuff I forgot.

The following stuff has not changed:
+ The stuff that shows up on your terminal when installing. Kept that exactly the same. 

Stuff that needs further disccussion:
~~+ The CLI interface has some mentions of skipping tool dependencies and installing resolver dependencies. Skipping tool dependencies and installing resolver and repository dependencies are now the default, so these options do not make sense. They are ignored in `def main` but kept for backwards compatibility. (I have broken every one's scripts already once... )~~

I hope it is much easier to work now on `shed_tools`. Should also be a lot easier to write a testing framework now using pytest #80 .

Hope this somewhat redeems myself for not being at the galaxy hackathon.

### CODE REVIEW

File: tests/data_manager_list.yaml [hunk 1]
```diff
@@ -7,4 +7,6 @@ tools:
   owner: devteam
 - name: data_manager_bwa_mem_index_builder
   owner: devteam
+  revisions:
+    - e519769cdb9c
```
Comment (mvdbeek) [inline tests/data_manager_list.yaml]: Can you try 46066df8813d as well, which is the latest version? I just tested this revision and it works, my hypothesis is that not specifying a revision doesn't install the latest revision after these changes.

Comment (rhpvorderman) [reply tests/data_manager_list.yaml]: Yeah sure, will try that.

Comment (mvdbeek) [reply tests/data_manager_list.yaml]: Do you have the log / commit somewhere ?

**Comment by rhpvorderman** [comment]
Added extra CLI functionality to solve #101

**Comment by mvdbeek** [comment]
I think I have implemented all those changes in https://github.com/rhpvorderman/ephemeris/pull/3

**Comment by rhpvorderman** [comment]
Thanks for all the changes @mvdbeek. Especially for fixing the error of get-tool-list listing deleted tools. I ran into this error, but did not really understand why it was happening, and then forget about it. Thanks for all the effort on this pull request!

> This is nice, I like it in principle. I'm ambivalent on the name change of install_repositories to install_tools, because we only install repositories, and I'm afraid that distinction is lost with these changes. Which is sort-of OK, because nothing here actually deals with tools, and we were not very consistent from the start. 

Yes this is quite ambivalent trough ephemeris. `get-tool-list` also gives you a list of repositories. So making this consistent across ephemeris would mean we have to use `get-repository-list`. Internally it can be made consistent though, to avoid confusion. 

Since we are going to break planemo anyway, I better do that now to avoid breaking it twice. Commits coming up.

**Comment by rhpvorderman** [comment]
Renamed all the tools that are actually repos to repos.

I kept the `test_tools` method, since it actually tests tools. It tests all the tools in a certain list of repos. So kept that the same. I hope it is not too confusing.

Also kept the user interface the same. I can rename everything in the documentation to repositories properly, and also change the information in the parser. But I do not think that is something we should do in this pull request.

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mvdbeek) [inline .idea/vcs.xml]: We'll want to put this file into .gitignore, right ?

Comment (rhpvorderman) [reply .idea/vcs.xml]: Whoops. Did put this in global .gitignore. But apparently got committed anyway.

Comment (rhpvorderman) [reply .idea/vcs.xml]: fixed now by recommitting changes

**Pull request merged**